.github/workflows/pr_test_cpu.yml

```diff
@@ -0,0 +1,54 @@
+name: Run tests on CPU
+
+on:
+  pull_request:
+    branches: [master]
+    types: [open, reopened, synchronize, review_requested, ready_for_review]
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.event.pull_request.number || github.ref }}
+  cancel-in-progress: true
+
+jobs:
+  tests-cpu:
+    name: ${{ matrix.os }} - python-${{ matrix.python-version }}, torch-${{ matrix.pytorch-version }}
+    runs-on: ${{ matrix.os }}
+
+    strategy:
+      fail-fast: true
+      matrix:
+        os: [ubuntu-latest, windows-latest, macos]
+        python-version: ['3.7', '3.8']
+        pytorch-version: ['1.9.1', '1.12.1']
```

Comment:
```suggestion
        pytorch-version: ['1.9.1', '1.12.1']
        pytorch-dtype: ["float16", "float32", "float64"]

Reply:
I think this very nice, but this will generate 36 jobs now and 72, after when we add python 39 and 310. Isn't that a lot?

Reply:
Also, trigger again the problem of waiting to get a windows or macos runner + setup each one (takes ~5min)

Reply:
depends on the time it takes to finish the whole thing we can reduce, probably we can exclude patterns. What's the solution for windows and mac?

Reply:
Ok, i will push this to test them

The only time-saving solution here that I see is to have your own runners, or just decrease the number of jobs on these platforms -- maybe just test max and min version of the supported python?

Reply:
For windows idk, Mac I let @ducha-aiki decide since he’s the main public developer using it